CMakeLists.txt

```diff
@@ -35,8 +31,7 @@ add_executable(seqgen sequence_generator_main.cpp)
 target_link_libraries(seqgen sequence util)
 
 # TESTS
-string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra -Werror")
-
+string(APPEND CMAKE_CXX_FLAGS " -Wall -Wextra -Werror  -msse4")
```

Comment:
You could consider adding `-fsanitize=address,undefined` here to enable some sanitizers. This will slow down build times quite a bit, but is super convenient to quickly determine the source of segfaults and such.

Reply:
would you only add it to the debug flags? or it won't affect the performance and can be added to release as well?

Reply:
Definitely don't put it on release.

For test, I'm not sure you'll always want it because compile times become *A LOT* slower. Maybe just add it as a comment or so so you can easily copy them in case you do want to use them. Anyway it's good to be aware of these.